pkg/pillar/cmd/domainmgr/metric.go

```diff
@@ -55,13 +59,56 @@ func getAndPublishMetrics(ctx *domainContext, hyper hypervisor.Hypervisor) {
 			continue
 		}
 		dm.UUIDandVersion.UUID = uuid
+		if !isXen {
+			// XXX from observation, the dm.CPUTotal gets from the kvm hypervisor containerd stat is
+			// hundreds percentage of the app uptime in seconds.
+			// Assume the CPUTotal value and the app CPU usage is proportional, this nonXenCPUfactor
+			// is obtained by run the same App under Xen and KVM in comparison to be used as
+			// an approcimation.
+			dm.CPUTotal = dm.CPUTotal / nonXenCPUfactor
+		}
 		ctx.pubDomainMetric.Publish(dm.Key(), dm)
 	}
 
 	hm, _ := hyper.GetHostCPUMem()
+	if !isXen {
+		// the the hypervisor other than Xen, we don't have the Dom0 stats. Get the host
+		// cpu and memory for the device here
+		formatAndPublishHostCPUMem(ctx, hm)
+	}
 	ctx.pubHostMemory.Publish("global", hm)
 }
 
+func formatAndPublishHostCPUMem(ctx *domainContext, hm types.HostMemory) {
+	cpuInfo, err := cpu.Get()
+	if err != nil {
+		log.Errorf("getAndPublishMetrics: cpu Get error %v", err)
+		return
+	}
+
+	used := hm.TotalMemoryMB - hm.FreeMemoryMB
+	var usedPerc float64
+	if hm.TotalMemoryMB > 0 {
+		usedPerc = float64(used * 100.0 / hm.TotalMemoryMB)
+	}
+	var hostUUID types.UUIDandVersion
+	hostUUID.UUID = nilUUID
+	uptime, _ := host.Uptime()
+	cpuUsage := cpuInfo.User + cpuInfo.System + cpuInfo.Iowait + cpuInfo.Softirq + cpuInfo.Irq
+	cpuUsageSec := (cpuUsage * uptime) / cpuInfo.Total
```

Comment:
I don't understand this. The cpuInfo.User, cpuInfo.System is already in a unit of seconds (or nanoseconds) which is what you need to report as CPUTotal.

Reply:
well the unit is called ticks, which is said to be in USER_HZ resolution. in our system, it is 100. So if I divide this cpuInfo.Total / USER_HZ, what I got was like the number about 4.5 larger than the device uptime in seconds. Thus I'm taking another way, knowing the relative percentage reported is correct, to let cloud side to get the percentage, i do this normalization here.

Reply:
I think it is better to use the gopsutil/cpu package we already use to remove one potential variable. Main difference seems to be that 1) it reports things as floats and 2) you can get individual counters per CPU

I think the Total number is bogus, because it it should always be uptime*ncpu, basically it doesn't represent the uptime.
What we care about is the sum of User and System across all the CPUs; that is what we need to report.

Reply:
gopsutil/cpu also takes care of the HZ to seconds conversion. Note that iowait and the various guest times do not count actual CPU usage used by the host. Thus the things we should add are

		busy := t.User + t.System + t.Nice + t.Irq + t.Softirq
where we get t for all CPUs using 
	cpuStat, err := cpu.Times(false)
	for _, t := range cpuStat {